[Core] Factorize two variables with the same meaning (constraint id)

alxbilger opened the pull request:
I suspect that `m_cId` (introduced in https://github.com/sofa-framework/sofa/commit/142aa4ba1f5299301633600ba8031e8159b67d55) and `m_constraintIndex` mean the same thing. I suggest to remove `m_cId` and to keep the Data.

Related to https://github.com/sofa-framework/sofa/pull/4763 and https://github.com/SofaDefrost/SoftRobots/pull/283



______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

alxbilger committed:
[Core] Factorize two variables with the same meaning (constraint id)

fredroy reviewed (approved):
I approve for the idea but I cannot really judge for the effects 😅

alxbilger committed:
add deprecation

alxbilger commented:
I added deprecation for the removed variable, and renamed the Data variable name (with deprecation). It will trigger warnings in SoftRobots and Cosserat.

alxbilger commented:
[ci-build][with-all-tests]

fredroy commented on Sofa/framework/Core/src/sofa/core/behavior/BaseConstraintSet.h:
```diff
@@ -92,11 +95,16 @@ class SOFA_CORE_API BaseConstraintSet : public virtual objectmodel::BaseObject
     Data< int > group; ///< ID of the group containing this constraint. This ID is used to specify which constraints are solved by which solver, by specifying in each solver which groups of constraints it should handle.
 public:
-    Data< unsigned int > m_constraintIndex; ///< Constraint index (first index in the right hand term resolution vector)
+    Data< unsigned int > d_constraintIndex; ///< Constraint index (first index in the right hand term resolution vector)
```
```suggestion
    Data< sofa::Index > d_constraintIndex; ///< Constraint index (first index in the right hand term resolution vector)
```

fredroy replied:
If I want to be picky 🫢

alxbilger committed:
change integer type

Co-authored-by: Frederick Roy <fredroy@users.noreply.github.com>

bakpaul commented:
[ci-build][force-full-build][with-all-tests]
